#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_sysctl_malloclen.dpatch by  <csmall@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adjust malloc length one more Closes: #423704

@DPATCH@
diff -urNad procps-3.2.7~/sysctl.c procps-3.2.7/sysctl.c
--- procps-3.2.7~/sysctl.c	2006-06-25 10:51:51.000000000 +1000
+++ procps-3.2.7/sysctl.c	2007-05-16 22:37:40.000000000 +1000
@@ -135,7 +135,7 @@
    }
 
    /* used to open the file */
-   tmpname = malloc(strlen(name)+strlen(PROC_PATH)+1);
+   tmpname = malloc(strlen(name)+strlen(PROC_PATH)+2);
    strcpy(tmpname, PROC_PATH);
    strcat(tmpname, name); 
    slashdot(tmpname+strlen(PROC_PATH),'.','/'); /* change . to / */
